build: update to rules_nodejs 0.38

alexeagle opened the pull request.

alexeagle committed:
build: update to rules_nodejs 0.38

alexeagle commented:
Caretaker: cl/271704225 needs to be patched in along with this change, so that the `.mjs` extensions remain `.closure.js` for google3

devversion committed:
ci: update material-unit-test commit

Updates the `material-unit-test` job to run tests against
the latest commit of the Angular Components repository.

We need to update to a version where components already
updated to `rules_nodejs#0.38.0`.

devversion committed:
ci: avoid postinstall error when building material-unit-tests job

The components repository updated to rules_nodejs#0.38.0 before Angular Bazel did.
To do this, the `@angular/bazel` v0.38.0 compatibility changes were patched on postinstall.

This now conflicts because we install a `@angular/bazel` version in the `material-unit-tests` job
that already includes these compatibility changes. This would result in the patch being a noop
for which the `patch` command throws an error.

We can remove this once components can install a released version of `@angular/bazel` that is
compatible with `rules_nodejs#0.38.0`.

alexeagle committed:
build: add missing http-server dep to bazel example

Also update rules_nodejs 0.38.0->0.38.1

gregmagolan committed:
build: add JSEcmaScripModuleInfo provider to ng_module

gregmagolan commented:
@googlebot I consent

gregmagolan commented on packages/bazel/src/builders/files/src/BUILD.bazel.template:
```diff
@@ -2,7 +2,7 @@ package(default_visibility = ["//visibility:public"])
 load("@npm_angular_bazel//:index.bzl", "ng_module")
 load("@npm_bazel_karma//:index.bzl", "ts_web_test_suite")
-load("@build_bazel_rules_nodejs//:defs.bzl", "rollup_bundle", "history_server")
+load("@build_bazel_rules_nodejs//:index.bzl", "rollup_bundle", "history_server")
```
Overlooked `history_server` here?

gregmagolan replied:
Done

gregmagolan committed:
build: add history-server to @angular/bazel schematics

The history_server rule is not longer shipped with rules_nodejs as it has been replaced by auto-generated rule `load("@npm//history-server:index.bzl", "history_server")` which requires the user to add history-server to their package.json.

josephperrott reviewed (approved):
LGTM

alexeagle commented:
CARETAKER: see note above about g3 CL needed to patch in with this

alxhub commented:
Note: I'm merging this to `master` only as it does not apply cleanly on the patch branch. @alexeagle will open a separate PR for patch-only.
